File: common/src/main/java/mods/cybercat/gigeresque/client/entity/render/AlienEntityRenderer.java
package mods.cybercat.gigeresque.client.entity.render;

import com.mojang.blaze3d.vertex.PoseStack;
import mod.azure.azurelib.rewrite.render.AzLayerRenderer;
import mod.azure.azurelib.rewrite.render.AzModelRenderer;
import mod.azure.azurelib.rewrite.render.entity.AzEntityRenderer;
import mod.azure.azurelib.rewrite.render.entity.AzEntityRendererConfig;
import mod.azure.azurelib.rewrite.render.entity.AzEntityRendererPipeline;
import mods.cybercat.gigeresque.client.entity.render.helper.ClassicModelRenderer;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.NotNull;

import mods.cybercat.gigeresque.Constants;
import mods.cybercat.gigeresque.client.entity.texture.EntityTextures;
import mods.cybercat.gigeresque.common.entity.animators.classic.ClassicAlienAnimator;
import mods.cybercat.gigeresque.common.entity.impl.classic.ClassicAlienEntity;

public class AlienEntityRenderer extends AzEntityRenderer<ClassicAlienEntity> {

    private static final ResourceLocation MODEL = Constants.modResource("geo/entity/alien/alien.geo.json");

    public AlienEntityRenderer(EntityRendererProvider.Context context) {
        super(
            AzEntityRendererConfig.<ClassicAlienEntity>builder($ -> MODEL, xeno -> {
                var progress = Math.max(0, Math.min(1 - (xeno.getGrowth() / xeno.getMaxGrowth()), 1));

                if (xeno.isPassedOut()) {
                    return EntityTextures.ALIEN_STATIS;
                }
                if (progress > 0) {
                    return EntityTextures.ALIEN_YOUNG;
                }
                return EntityTextures.ALIEN;
            })
                .setAnimatorProvider(ClassicAlienAnimator::new)
                .setDeathMaxRotation(0.0F)
                .build(),
            context
        );
        this.shadowRadius = 0.5f;
    }

    @Override
    protected AzEntityRendererPipeline<ClassicAlienEntity> createPipeline(AzEntityRendererConfig<ClassicAlienEntity> config) {
        return new AzEntityRendererPipeline<>(config, this) {

            @Override
            protected AzModelRenderer<ClassicAlienEntity> createModelRenderer(AzLayerRenderer<ClassicAlienEntity> layerRenderer) {
                return new ClassicModelRenderer(this, layerRenderer);
            }
        };
    }

    /*
     * Remove scale once the Rom Alien is implemented
     */
    @Override
    public void render(
        ClassicAlienEntity entity,
        float entityYaw,
        float partialTick,
        PoseStack stack,
        @NotNull MultiBufferSource bufferSource,
        int packedLightIn
    ) {
        var scaleFactor = 0.8f + ((entity.getGrowth() / entity.getMaxGrowth()) / 5f);
        stack.scale(scaleFactor, scaleFactor, scaleFactor);
        super.render(entity, entityYaw, partialTick, stack, bufferSource, packedLightIn);
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/AlienEntityRenderer.kt
package com.bvanseg.gigeresque.client.entity.render

import com.bvanseg.gigeresque.client.entity.model.AlienEntityModel
import com.bvanseg.gigeresque.client.entity.render.feature.ClassicAlienFeatureRenderer
import com.bvanseg.gigeresque.client.extensions.scale
import com.bvanseg.gigeresque.common.entity.impl.ClassicAlienEntity
import net.minecraft.client.render.VertexConsumerProvider
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.util.math.MatrixStack
import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer

/**
 * @author Boston Vanseghi
 */
class AlienEntityRenderer(context: EntityRendererFactory.Context) : GeoEntityRenderer<ClassicAlienEntity>(
    context,
    AlienEntityModel()
) {
    init {
        this.shadowRadius = 0.5f
        this.addLayer(ClassicAlienFeatureRenderer(this))
    }

    override fun render(
        entity: ClassicAlienEntity,
        entityYaw: Float,
        partialTicks: Float,
        stack: MatrixStack,
        bufferIn: VertexConsumerProvider,
        packedLightIn: Int
    ) {
        stack.scale(0.5f + ((entity.growth / entity.maxGrowth.toFloat()) / 5f))
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
    }
}